pkg/gofr/http/errors.go

```diff
@@ -0,0 +1,76 @@
+package http
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// ErrorEntityNotFound represents an error for when an entity is not found in the system.
+type ErrorEntityNotFound struct {
+	FieldName  string
+	FieldValue string
+}
+
+// ErrorInvalidParam represents an error for invalid parameter values.
+type ErrorInvalidParam struct {
+	Param []string `json:"param,omitempty"` // Param contains the list of invalid parameter names.
+}
+
+// ErrorMethodNotAllowed represents an error for when a method is not allowed on a URL.
+type ErrorMethodNotAllowed struct {
+	Method string `json:"method"`
+	URL    string `json:"url"`
+}
```

Comment:
we can remove this error, as gofr maps the route and users might not need the following error, may be used for internal purpose if required

Reply:
sure will remove